src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -2523,6 +2530,63 @@ class CreateStatementSegment(BaseSegment):
         ),
         Ref("IfNotExistsGrammar", optional=True),
         Ref("ObjectReferenceSegment"),
+        # Next set are Storage Integration statements
+        # https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html
+        Sequence(
+            AnySetOf(
+                Sequence("TYPE", Ref("EqualsSegment"), "EXTERNAL_STAGE"),
+                Sequence("ENABLED", Ref("EqualsSegment"), Ref("BooleanLiteralGrammar")),
+                OneOf(
+                    Ref("S3StorageIntegrationParameters"),
+                    Ref("GCSStorageIntegrationParameters"),
+                    Ref("AzureStorageIntegrationParameters"),
+                ),
+                Sequence(
+                    "STORAGE_ALLOWED_LOCATIONS",
+                    Ref("EqualsSegment"),
+                    OneOf(
+                        Bracketed(
+                            Delimited(
+                                OneOf(
+                                    Ref("S3Path"),
+                                    Ref("GCSPath"),
+                                    Ref("AzureBlobStoragePath"),
+                                )
+                            )
+                        ),
+                        Bracketed(
+                            Ref("QuotedStarSegment"),
+                        ),
+                    ),
+                ),
+                Sequence(
+                    "STORAGE_BLOCKED_LOCATIONS",
+                    Ref("EqualsSegment"),
+                    OneOf(
+                        Bracketed(
+                            Delimited(
+                                OneOf(
+                                    Ref("S3Path"),
+                                    Ref("GCSPath"),
+                                    Ref("AzureBlobStoragePath"),
+                                )
+                            )
+                        ),
+                        Bracketed(
+                            Ref("QuotedStarSegment"),
+                        ),
+                    ),
+                    optional=True,
+                ),
+                Sequence(
+                    "COMMENT",
+                    Ref("EqualsSegment"),
+                    Ref("QuotedLiteralSegment"),
+                    optional=True,
+                ),
+            ),
+            optional=True,
```

Comment:
Ah, I was going from the docs where it has this syntax:

```
...
  ENABLED = { TRUE | FALSE }
  STORAGE_ALLOWED_LOCATIONS = ('<cloud>://<bucket>/<path>/', '<cloud>://<bucket>/<path>/')
  [ STORAGE_BLOCKED_LOCATIONS = ('<cloud>://<bucket>/<path>/', '<cloud>://<bucket>/<path>/') ]
```

so I think it comes down to the parse versus. runtime error issue.

With the above, `STORAGE_ALLOWED_LOCATIONS` is required - but it can come before the other arguments (e.g.) `ENABLED` - whereas `STORAGE_BLOCKED_LOCATIONS` is optional.

I suppose what I'm really looking for is an `UnorderedSequence()` rather than `AnySetOf()` - I didn't actually realise that `AnySetOf()` is implicitly optional.

Is there anything like `UnorderedSequence` available?

If not - I'm happy to do your suggestions of removing the `optional=True` from these. Let me know!

Reply:
Don't think we can really handle that to be honest. So think we need to basically treat both as optional.

Reply:
Sure thing, I'll start to make the above changes :)

Reply:
I still don't think we need this `optional=True`

Reply:
Ah yes. This was me IS needed!